### PR TITLE
[X86][MC] Support enc/dec for SETZUCC and promoted SETCC.

### PR DESCRIPTION
**Pull request by FreddyLeaf**
apx-spec: https://cdrdv2.intel.com/v1/dl/getContent/784266
apx-syntax-recommendation: https://cdrdv2.intel.com/v1/dl/getContent/817241

### CODE REVIEW

File: llvm/test/MC/Disassembler/X86/apx/setcc-evex.txt
Comment (KanRobert) [inline llvm/test/MC/Disassembler/X86/apx/setcc-evex.txt:1]: Remove "-evex" from the file name. Same for others.

Comment (FreddyLeaf) [reply llvm/test/MC/Disassembler/X86/apx/setcc-evex.txt:1]: 85f8ab0b7986b8e1f823ac4cc5b639440b0ce77c

File: llvm/lib/Target/X86/X86InstrCMovSetCC.td [hunk 1]
```diff
@@ -127,6 +127,23 @@ let Uses = [EFLAGS], isCodeGenOnly = 1, ForceDisassemble = 1 in {
                 TB, Sched<[WriteSETCCStore]>;
 } // Uses = [EFLAGS]
 
+// SetZUCC and promoted SetCC instructions.
+let Uses = [EFLAGS], mayStore = 1, isCodeGenOnly = 1, ForceDisassemble = 1,
```
Comment (KanRobert) [inline llvm/lib/Target/X86/X86InstrCMovSetCC.td:131]: Incorrect. `mayStore=1` should be set for memory variant only.

File: llvm/test/MC/X86/apx/setcc-att.s [hunk 2]
```diff
@@ -0,0 +1,98 @@
+# RUN: llvm-mc -triple x86_64 -show-encoding %s | FileCheck %s
+
```
Comment (KanRobert) [inline llvm/test/MC/X86/apx/setcc-att.s:2]: check error in 32-bit mode

File: llvm/test/MC/X86/apx/setzucc-att.s [hunk 3]
```diff
@@ -0,0 +1,98 @@
+# RUN: llvm-mc -triple x86_64 -show-encoding %s | FileCheck %s
+
```
Comment (KanRobert) [inline llvm/test/MC/X86/apx/setzucc-att.s:2]: ditto

**Review by KanRobert — APPROVED** [review]
LGTM

**Pull request merged**